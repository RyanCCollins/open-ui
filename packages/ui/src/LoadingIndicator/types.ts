import * as React from 'react';
import LoadingIndicator from './';

export interface LoadingIndicatorProps extends React.Props<typeof LoadingIndicator> {
  isLoading: boolean;
}
