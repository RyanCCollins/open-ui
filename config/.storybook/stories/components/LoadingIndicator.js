import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { LoadingIndicator } from 'ui';
import {
  withKnobs,
  boolean
} from '@kadira/storybook-addon-knobs';

const stories = storiesOf('Loading Indicator', module);
stories.addDecorator(withKnobs);

stories
  .addWithInfo(
    'is loading',
    `
      Properties:

      \`isLoading: true|false\`
    `,
     () =>
      <LoadingIndicator
        isLoading={boolean('isLoading','true')}
      />,
    {
      inline: true,
      propTables: false,
    }
  );
