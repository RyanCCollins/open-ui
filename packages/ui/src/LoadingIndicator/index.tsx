// Adapted from https://grommet.github.io/docs/spinning
import * as React from 'react';
import { Box } from '../';
import { LoadingIndicatorProps } from './types';
import SvgIcon, { Circle } from './styles';

export default function LoadingIndicator({
  isLoading,
}: LoadingIndicatorProps): JSX.Element {
  if (!isLoading) {
    return null;
  }
  return (
    <Box
      boxSize={{ horizontal: 'small' }}
      pad="medium"
      alignItems="center"
      justifyContent="center"
    >
      <SvgIcon viewBox="0 0 48 48">
        <Circle
          cx="24"
          cy="24"
          r="21"
          stroke="#007acc"
          strokeWidth="6"
          fill="none"
        />
      </SvgIcon>
    </Box>
  );
};
