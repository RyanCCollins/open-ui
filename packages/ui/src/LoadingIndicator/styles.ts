const styled = require('styled-components').default;
const { keyframes } = require('styled-components');

const draw = keyframes`
  0% {
    transform: rotate(-90deg);
    stroke-dashoffset: 151px;
  }

  100% {
    stroke-dashoffset: -131px
  }
`;

const color = keyframes`
  0% {
    stroke: #007acc;
  }

  100% {
    stroke: #293953;
  }
`;

export const Circle = styled.circle`
  animation: ${color} 1s alternate infinite ease-in-out;
`;

export default styled.svg`
  width: 48px;
  height: 48px;
  stroke-dasharray: 151px 151px;
  stroke-dashoffset: 0;
  transform: rotate(270deg);
  animation: ${draw} 4s alternate infinite ease-in-out;
`;
