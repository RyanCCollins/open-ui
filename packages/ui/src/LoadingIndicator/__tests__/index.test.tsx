import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import LoadingIndicator from '../';

describe('<LoadingIndicator /> default', () => {
  it('should render the element when isLoading is true', () => {
    const wrapper = shallow(
      <LoadingIndicator isLoading={true} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should return null when isLoading is false', () => {
    const wrapper = shallow(
      <LoadingIndicator isLoading={false} />,
    );
    expect(shallowToJson(wrapper)).toBeNull();
  });
});
