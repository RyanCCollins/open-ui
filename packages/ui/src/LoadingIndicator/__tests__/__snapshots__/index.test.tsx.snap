// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> default should render the element when isLoading is true 1`] = `
<Box
  alignItems="center"
  boxSize={
    Object {
      "horizontal": "small",
    }
  }
  justifyContent="center"
  pad="medium"
>
  <styled.svg
    viewBox="0 0 48 48"
  >
    <styled.circle
      cx="24"
      cy="24"
      fill="none"
      r="21"
      stroke="#007acc"
      strokeWidth="6"
    />
  </styled.svg>
</Box>
`;
